release.py

```diff
@@ -413,28 +415,29 @@ def update_uv_lock():
     subprocess.run(['uv', 'lock'], check=True)  # noqa: S607
 
 
-def parse_scenario_version() -> tuple[str, str]:
+def parse_scenario_version() -> packaging.version.Version:
     """Parse the current scenario version from pyproject.toml."""
     file_path = VERSION_FILES['testing']
     content = file_path.read_text()
     match = re.search(rf'version = "{VERSION_REGEX}"', content)
     if not match:
         raise ValueError(f'Could not find version string in {file_path}')
     version_str = match.group(1)
-    base_version, dev_suffix = parse_version(version_str)
-    return base_version, dev_suffix
+    return parse_version(version_str)
 
 
 def get_new_scenario_version(ops_version: str) -> str:
-    """Get a new version for scenario based on ops verwsion.
+    """Get a new version for scenario based on ops version.
 
     We want the scenario version to always be exactly ops major version+5,
     like ops 3.1.2 -> scenario 8.1.2, and ops 3.1.2.dev0 -> scenario 8.1.2.dev0.
+    Pre-release versions are also preserved, like ops 3.1.2b1 -> scenario 8.1.2b1.
     """
-    base_version, dev_suffix = parse_version(ops_version)
-    major, minor, patch = base_version.split('.')
-    dev = dev_suffix if dev_suffix else ''
-    return f'{int(major) + 5}.{minor}.{patch}{dev}'
+    version = parse_version(ops_version)
+    major, minor, patch = version.release
+    pre_suffix = f'{version.pre[0]}{version.pre[1]}' if version.pre else ''
+    dev_suffix = f'.dev{version.dev}' if version.dev is not None else ''
+    return f'{major + 5}.{minor}.{patch}{pre_suffix}{dev_suffix}'
```

Comment:
This feels kind of nitpicky as I type it out, but the `pre_suffix` line gave me pause. I wonder if using the slightly more verbose `+=` approach might be easier to follow here.
```suggestion
    new_version = f'{major + 5}.{minor}.{patch}'
    if version.pre is not None:  # then it's in the form ('a', 0)
        new_version += f'{version.pre[0]}{version.pre[1]}'
    if version.dev is not None:  # then it's the dev version number
        new_version += f'.dev{version.dev}'
    return new_version
```
Feel free to reject if you prefer the current form.